raiden/ui/cli.py

```diff
@@ -942,27 +943,37 @@ def _run_app():
     # TODO:
     # - Ask for confirmation to quit if there are any locked transfers that did
     # not timeout.
-    if kwargs['transport'] == 'udp':
-        (listen_host, listen_port) = split_endpoint(kwargs['listen_address'])
-        try:
-            with SocketFactory(listen_host, listen_port, strategy=kwargs['nat']) as mapped_socket:
-                kwargs['mapped_socket'] = mapped_socket
-                app_ = _run_app()
+    try:
+        if kwargs['transport'] == 'udp':
+            (listen_host, listen_port) = split_endpoint(kwargs['listen_address'])
+            try:
+                with SocketFactory(
+                        listen_host, listen_port, strategy=kwargs['nat'],
+                ) as mapped_socket:
+                    kwargs['mapped_socket'] = mapped_socket
+                    app_ = _run_app()
 
-        except RaidenServicePortInUseError:
-            print(
-                'ERROR: Address %s:%s is in use. '
-                'Use --listen-address <host:port> to specify port to listen on.' %
-                (listen_host, listen_port),
-            )
-            sys.exit(1)
-    elif kwargs['transport'] == 'matrix':
-        kwargs['mapped_socket'] = None
-        app_ = _run_app()
-    else:
-        # Shouldn't happen
-        raise RuntimeError(f"Invalid transport type '{kwargs['transport']}'")
-    app_.stop(leave_channels=False)
+            except RaidenServicePortInUseError:
+                print(
+                    'ERROR: Address %s:%s is in use. '
+                    'Use --listen-address <host:port> to specify port to listen on.' %
+                    (listen_host, listen_port),
+                )
+                sys.exit(1)
+        elif kwargs['transport'] == 'matrix':
+            kwargs['mapped_socket'] = None
+            app_ = _run_app()
+        else:
+            # Shouldn't happen
+            raise RuntimeError(f"Invalid transport type '{kwargs['transport']}'")
+        app_.stop(leave_channels=False)
+    except ReplacementTransactionUnderpriced:
```

Comment:
What happens if this is thrown via the RestAPI? How would we handle it there? Or it is not possible for the REST API to throw this exception?

The only way I can imagine this happening is what Paul describes [below](https://github.com/raiden-network/raiden/pull/1862/files#r203328554).

But unless I am mistaken we don't really retry transactions for blockchain transactions that were sent through the state machine.

For the ones that are not sent through the state machine, like [`channel_open()`](https://github.com/raiden-network/raiden/blob/d38d2a3b20b82a77d6adc1369e7550929d729886/raiden/api/python.py#L186) we [keep all transactions](https://github.com/raiden-network/raiden/blob/d38d2a3b20b82a77d6adc1369e7550929d729886/raiden/network/proxies/token_network.py#L137) sent by the node to the chain and still not resend them.

Reply:
I think it depends on API and how it's implemented right? if we know that this error would be triggered by sending 2 transactions to the blockchain with the same nonce and <= gas amount. But in general, what we wanted to achieve with this PR is to warn about a strange behavior which shouldn't happen normally if the raiden node had exclusive access to the account.

Reply:
> What happens if this is thrown via the RestAPI

The `RestAPI` is executed inside the [`_run_app()` call](https://github.com/rakanalh/raiden/blob/28ef251613351c0807d885c14ca64da0a69e00b4/raiden/ui/cli.py#L908), so this try-except will caught exceptions thrown there too